repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -418,6 +565,105 @@ private void traverseEdgesOnDemand(AtlasVertex datasetVertex, boolean isInput, i
         }
     }
 
+    private boolean calculateLineageSize(AtlasVertex baseDatasetVertex, boolean isInput, AtlasLineageSizeContext context, int currentDepth, AtomicInteger currentCount) {
```

Comment:
Lets not use boolean flags for input and output, rather pass the enums to be on safer side and better readability

Reply:
Have followed this pattern since all lineage APIs were written this way. Just to keep code similar.